ocean_lib/ocean/ocean_assets.py

```diff
@@ -322,48 +375,185 @@ def create(
+        :param publisher_wallet: Wallet of the publisher updating this asset.
+        :param delete_erc20_tokens: List of existing ERC20 tokens and related Services to delete.
+        :param encrypted_files: str of the files that need to be encrypted before updating.
+        :param new_services: list of new Service objects to add.
+        :param new_credentials: new_credentials necessary for the asset.
+        :param provider_uri: str URL of service provider. This will be used as base to construct the serviceEndpoint for the `access` (download) service
+        :param new_erc20_tokens_data: list of ERC20CreateData necessary for deploying ERC20 tokens for different services.
+        :param new_deployed_erc20_tokens: list of ERC20 tokens which are already deployed.
+        :param encrypt_flag: bool for encryption of the DDO.
+        :param compress_flag: bool for compression of the DDO.
+        :return: update tx_result
+        """
+
+        old_asset = self._get_aquarius(self._metadata_cache_uri).wait_for_asset(did)
+
+        assert old_asset, "Asset not found."
+
+        if new_deployed_erc20_tokens or new_erc20_tokens_data:
+            assert encrypted_files, "Encrypted_files is required for adding new tokens."
+
+        # Check if datatokens in delete_erc20_tokens are in the old_asset
+        for delete_erc20_tokens_data in delete_erc20_tokens:
+            assert delete_erc20_tokens_data in [
+                dt.get("address") for dt in old_asset.datatokens
+            ], f"{delete_erc20_tokens_data} to delete is not in the old asset datatokens."
+
+        # Metadata sanity check
+        assert isinstance(
+            new_metadata, dict
+        ), f"Expected new_metadata of type dict, got {type(new_metadata)}"
+
+        asset_type = new_metadata.get("type")
+        assert asset_type in (
+            "dataset",
+            "algorithm",
+        ), f"Invalid/unsupported asset type {asset_type}"
+
+        assert "name" in new_metadata, "Must have name in new_metadata."
+
+        if not provider_uri:
+            provider_uri = DataServiceProvider.get_url(self._config)
+
+        address = get_address_of_type(self._config, ERC721FactoryContract.CONTRACT_NAME)
+        erc721_factory = ERC721FactoryContract(self._web3, address)
+        erc721_address = old_asset.nft_address
+
+        # Verify nft address
+        if not erc721_factory.verify_nft(erc721_address):
+            raise ContractNotFound(
+                f"NFT address {erc721_address} is not found in the ERC721Factory events."
             )
-            document = encrypt_response.text
 
-        # Encrypted & compressed
+        assert erc721_address, "nft_address is required for publishing a dataset asset."
+        erc721_token = ERC721Token(self._web3, erc721_address)
+
+        # Copy original asset
+        asset = copy.deepcopy(old_asset)
+
+        # Check that the ddo is registered
+        if not self._get_aquarius().ddo_exists(did):
```

Comment:
you just got this asset from aquarius, there's no point in checking again if it exists.